ext/js/lib/js.rb

```diff
@@ -90,7 +91,10 @@ def self.__eval_async_rb(rb_code, future)
 
 class JS::Object
   def method_missing(sym, *args, &block)
-    if self[sym].typeof == "function"
+    if sym == :new
+      # Call constructor at the JavaScript runtime when new method is called
+      JS.eval("return #{self.to_construct(args)}")
```

Comment:
Can we simply define `def new` method under `JS::Object`?

Reply:
I have defined a new method.

Reply:
Thanks!